File: backend/app/routes/tactics.py
from flask import Blueprint, jsonify, request
from flask_jwt_extended import jwt_required
from app.services import tactic_service

bp = Blueprint("tactics", __name__, url_prefix="/api/tactics")


@bp.get("/")
@jwt_required()
def list_tactics():
    framework = request.args.get("framework")
    return jsonify({"data": tactic_service.list_tactics(framework=framework)})

File: frontend/src/api/tactics.js
import client from "./client";

export const listTactics = (params) => client.get("/api/tactics/", { params }).then((r) => r.data);
export const refreshMitre = (framework) =>
  client.post("/api/mitre/refresh", framework ? { framework } : {}).then((r) => r.data);
